tests/model_inheritance/models.py

```diff
@@ -186,3 +186,31 @@ class Base(models.Model):
 
 class SubBase(Base):
     sub_id = models.IntegerField(primary_key=True)
+
+
+#
+# test that uniques are checked for all parents
+
+@python_2_unicode_compatible
+class GrandParent(models.Model):
+    first_name = models.CharField(max_length=80)
+    last_name = models.CharField(max_length=80)
+    email = models.EmailField(unique=True)
+
+    class Meta:
+        unique_together = ('first_name', 'last_name')
+
+    def __str__(self):
+        return '"%s %s" <%s>' % (self.first_name, self.last_name, self.email)
```

Comment:
oh, its not used.
On Jan 30, 2015 5:47 PM, "Simon Charette" <notifications@github.com> wrote:

> In tests/model_inheritance/models.py
> <https://github.com/django/django/pull/3968#discussion_r23878432>:
>
> > +
> > +
> > +#
> > +# test that uniques are checked for all parents
> > +
> > +@python_2_unicode_compatible
> > +class GrandParent(models.Model):
> > +    first_name = models.CharField(max_length=80)
> > +    last_name = models.CharField(max_length=80)
> > +    email = models.EmailField(unique=True)
> > +
> > +    class Meta:
> > +        unique_together = ('first_name', 'last_name')
> > +
> > +    def __str__(self):
> > +        return '"%s %s" <%s>' % (self.first_name, self.last_name, self.email)
>
> Is this method used anywhere in the tests?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/django/django/pull/3968/files#r23878432>.
>